src/app/auth/auth.guard.ts

```diff
@@ -1,8 +1,19 @@
-import { CanActivateFn } from '@angular/router';
+import { inject } from '@angular/core';
+import { CanActivateFn, Router } from '@angular/router';
+import { AuthService } from './auth.service';
+import { JwtHelperService } from '@auth0/angular-jwt'
 
 export const authGuard: CanActivateFn = (route, state) => {
-  if (localStorage.getItem('token')){
+  const authService = inject(AuthService);
+  const token = authService.getToken();
+  const jwtHelper = new JwtHelperService();
+
+  if (token && !jwtHelper.isTokenExpired(token)){
```

Comment:
Token check shouldn't be there, because the responsibility of this guard only to check if the user is logged in (token exists only). The expiration token check should be in Http Interceptor (you can remove expiration tken check for now and we will add this logic later)